### DISCUSSION TITLE
Chat context

### DISCUSSION THREAD

**Discussion by asmagen** [nhaouari/obsidian-textgenerator-plugin#137] [General]
I understand what context is being used for a given query based on the cursor position or selected text, but are the previous queries and responses from the same session or document being used to inform the response to followup queries?

**Comment by nhaouari** [comment]
New empty line is used to consider the whole document as context.

**Reply by dvansoye** [reply]
How could this be used in the context of using templates? Is it possible to have a follow-on question within a template? I think this is called chained reasoning. Is it possible to set this up with a template? The reason I ask is because I am experimenting with reflection and the answers get quite long. If I could set it up as a discussion amongst multiple experts and call on each one of them I think it's likely I could get a comprehensive answer. Otherwise, I run out of tokens and the discussion just ends.
